src/main/java/net/minecraftforge/java_provisioner/GradleLocator.java

```diff
@@ -36,99 +37,71 @@ public class GradleLocator extends JavaHomeLocator {
     private static final Pattern GRADLE_ENV = Pattern.compile("JDK\\d\\d*");
 
     @Override
-    public File find(int version) {
-        List<IJavaInstall> results = new ArrayList<>();
-        fromGradleEnv(results, version);
-        if (!results.isEmpty())
-            return results.get(0).home();
+    public List<JavaInstall> findAll(int version) {
+        List<@Nullable JavaInstall> ret = new ArrayList<>();
```

Comment:
This is a perfect example of what I meant, you're searching for things when you could early exit on the first found result. Thats why I wrote it the way i did.